src/raw/client.rs

```diff
@@ -953,4 +1016,63 @@ mod tests {
         );
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_raw_scan_retryer() -> Result<()> {
```

Comment:
This test case looks a little not necessary to me, as it just check that we call error handler on region error, but introduce more complexity.

The core changes in this PR is the retry on region error, so if we want to verify the correctness, I think we need to simulate such scene.

For example, we first put some little entries, and perform the scan to fill the region cache. Then we put much more (or large) entries to trigger the region split (see `tests::common::init()`), and scan again. The later scan should meet region errors.

The test case would be a little complex to implement, so it's OK to me if we don't have it in this PR.

Reply:
Good point. The current setup, to your point, is a bit cumbersome. I have removed this test case. I will add the test as you outlined in a subsequent PR.